src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentNodeWriter.cs

```diff
@@ -165,7 +165,7 @@ protected void WriteComponentTypeInferenceMethod(CodeRenderingContext context, C
             switch (parameter.Source)
             {
                 case ComponentAttributeIntermediateNode attribute:
-                    context.CodeWriter.WriteStartInstanceMethodInvocation(ComponentsApi.RenderTreeBuilder.BuilderParameter, ComponentsApi.RenderTreeBuilder.AddAttribute);
+                    context.CodeWriter.WriteStartInstanceMethodInvocation(ComponentsApi.RenderTreeBuilder.BuilderParameter, ComponentsApi.RenderTreeBuilder.AddComponentParameter);
```

Comment:
Do we to do this conditionally based on the version of the runtime we're generating for?

Reply:
Good catch, I think we should do that. I guess RazorLanguageVersion is used for that (seen in https://github.com/dotnet/razor-compiler/pull/70). Added conditional checks (and restored the `object` casting in older versions).